opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -173,8 +173,7 @@ where
 
         let mut log_record = self.logger.create_log_record();
 
-        // TODO: Fix heap allocation
-        log_record.set_target(target.to_string());
+        log_record.set_target(target);
```

Comment:
oh wait.. this works now?

Reply:
@utpilla helped me understand better. We were unnecessarily copying the target before!

@gruebel Please do this change in a separate PR, with a before/after benchmark - it should show measurable improvement.